### PR TITLE
Pfappserver Network Visualization

### PR DESCRIPTION
**Pull request by satkunas**
# Description
Live network visualization using d3 force directed graph.

# Impacts
TBD

# Code / PR Dependencies
TBD

# NEW Package(s) required
TBD

# Issue
n/a

# Delete branch after merge
YES 

# Checklist
- [ ] Document the feature
- [n/a] Add unit tests
- [n/a] Add acceptance tests (TestLink)

# NEWS file entries
## New Features
* Live network visualization added to dashboard

# UPGRADE file entries
TBD

### CODE REVIEW

**Comment by extrafu** [comment]
image

**Comment by cgx** [comment]
Add the [d3-scale](https://github.com/d3/d3-scale) dependency to the `package.json` file.

**Comment by cgx** [comment]
Add the [typeface-b612-mono](https://hail2u.github.io/mn/b612-mono/) font dependency to the `package.json` file.

File: html/pfappserver/root/static.alt/src/components/pfNetworkGraphTooltipSwitch.vue [hunk 1]
```diff
@@ -75,10 +48,52 @@ export default {
...
+    return {
+      switche: false,
+      isLoading: false,
+      isError: false
+    }
+  },
+  methods: {
+    init () {
+      this.isLoading = true
+      apiCall.getQuiet(`config/switch/${this.id}`).then(response => {
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfNetworkGraphTooltipSwitch.vue]: Could we avoid an API call when `this.id` is `unknown`?

Comment (cgx) [reply html/pfappserver/root/static.alt/src/components/pfNetworkGraphTooltipSwitch.vue]: I guess we could benefit from a vuex store here to do some caching and speedup the tooltips.

**Comment by cgx** [comment]
@satkunas Even though *d3-force* is installed because it's a dependency of *plotly.js*, I would explicitly list the module in `package.json`.

File: html/pfappserver/root/static.alt/src/components/pfNetworkGraphTooltipNode.vue [hunk 2]
```diff
@@ -1,22 +1,18 @@
...
-                <p class="text-nowrap" v-t="'Not found'"></p>
-              </b-media>
-            </b-col>
+          <b-col cols="12" md="auto" class="w-100 text-center">
+            <icon name="circle-notch" scale="2" spin></icon>
+          </b-col>
         </b-row>
       </b-container>
-      <b-container fluid class="px-0" v-else>
+      <b-container fluid class="container px-0" v-else-if="!isError">
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfNetworkGraphTooltipNode.vue]: No need to use a *fluid* container for tooltips. I would simply use `<b-container class="px-0" ..>`.

File: html/pfappserver/root/static.alt/src/components/pfNetworkGraphTooltipPacketfence.vue [hunk 3]
```diff
@@ -1,25 +1,9 @@
 <template>
   <b-card no-body>
     <b-card-header>
-      <h5 class="mb-0"><mac>Packetfence</mac></h5>
+      <h5 class="mb-0">Packetfence</h5>
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfNetworkGraphTooltipPacketfence.vue]: `s/Packetfence/PacketFence`

File: html/pfappserver/root/static.alt/src/views/Status/_components/Network.vue [hunk 4]
```diff
@@ -47,27 +39,74 @@
...
+
+            <b-input-group class="mb-0" size="sm">
+              <b-input-group-prepend is-text>
+                <pf-form-range-toggle class="inline mt-n2" v-model="liveMode" :disabled="isLoading || !liveModeAllowed"></pf-form-range-toggle>
+              </b-input-group-prepend>
+              <b-dropdown variant="light" :text="$t('Live View')" :disabled="isLoading || !liveModeAllowed">
+                <b-dropdown-item v-for="timeout in [5000, 10000, 15000, 30000, 60000, 120000, 300000]" :key="timeout"
+                  :active="liveMode === true && liveModeIntervalMs === timeout"
+                  @click="liveMode = true; liveModeIntervalMs = timeout"
+                >{{ $t('{duration} seconds', { duration: timeout / 1E3 }) }}</b-dropdown-item>
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Status/_components/Network.vue:84]: You are a nerd 🤓

File: html/pfappserver/root/static.alt/src/components/pfNetworkGraphTooltipSwitch.vue [hunk 5]
```diff
@@ -75,10 +48,57 @@ export default {
...
+      switche: false,
+      isLoading: false,
+      isError: false
+    }
+  },
+  methods: {
+    init () {
+      if (this.id !== 'unknown') {
+        this.isLoading = true
+        apiCall.getQuiet(`config/switch/${this.id}`).then(response => {
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfNetworkGraphTooltipSwitch.vue:62]: Caching switch properties into a Vuex store would be nice.

File: html/pfappserver/root/static.alt/src/components/pfNetworkGraph.vue [hunk 6]
```diff
@@ -604,20 +521,221 @@ export default {
...
+            const { depth, num } = node
+            let offset
+            let angle
+            let distance
+            switch (type) {
+              case 'node':
+                offset = this.sortedNodes.filter(n => n.id === id).reduce((offset, node) => {
+                  let { id, source = {} } = node
+                  do {
+                    if (source && 'targets' in source) {
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfNetworkGraph.vue]: How about extracting *targets* from *source* when destructuring *node*?

File: html/pfappserver/root/static.alt/src/components/pfNetworkGraph.vue [hunk 7]
```diff
@@ -1336,7 +1347,10 @@ export default {
         border-style: solid;
         border-width: 2px;
         box-shadow: 5px 5px 15px rgba(0, 0, 0, 0.125);
+        transform: translate(-50%, -50%);
         z-index: 4;
+        max-width: 175px;
+        width: 175px;
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfNetworkGraph.vue]: Maybe we could use *vw* units here?

**Pull request merged**